test/unit/specs/core/network/createNetworkStrategy.spec.js

```diff
@@ -186,6 +190,43 @@ describe("networkStrategyFactory", () => {
               .catch(done.fail);
           });
       });
+
+      whenMockServerIsRunningIt("sends a large beacon", done => {
+        client
+          .mockAnyResponse({
+            httpRequest: {
+              method: "POST",
+              path: "/myapi"
+            },
+            httpResponse: {
+              statusCode: 204
+            }
+          })
+          .then(() => {
+            networkStrategy(
+              "http://localhost:1080/myapi",
+              largeRequestBody,
+              true
+            )
+              .then(body => {
+                expect(body).toBeUndefined();
+                // no setTimeout here like for the regular beacon test because the promise returned
+                // should be the fetch promise that is resolved when it was successfully sent.
+                client
+                  .verify(
+                    {
+                      method: "POST",
+                      path: "/myapi",
+                      body: largeRequestBody
+                    },
+                    1,
+                    1
+                  ) // verify atLeast 1, atMost 1 times called
+                  .then(done, done.fail);
+              })
+              .catch(done.fail);
+          });
+      });
```

Comment:
I don't think there's any need to use `done` in these tests. Just return the resulting promise from the promise chain. If the promise gets resolved, Jasmine will consider it passed. If the promise gets rejected, Jasmine will consider it failed.